### PR TITLE
feat: clearing all messages with same group id when the batch processing fails FGR3-4960

### PR DESCRIPTION
**Pull request by zabkwak**

### CODE REVIEW

File: src/consumer.ts [hunk 1]
```diff
@@ -389,8 +391,14 @@ export class Consumer extends EventEmitter {
             } else {
                 err.message = `Unexpected message handler failure: ${err.message}`
             }
+            const [message] = messages
```
Comment (Crhaj) [inline src/consumer.ts:394]: This is weird and I don't think it is very clear to everyone what does it do... I would at least do const [message,] = messages

File: tests/consumer.spec.ts [hunk 2]
```diff
@@ -989,6 +989,7 @@ describe("Consumer", () => {
             sandbox.assert.calledTwice(handleMessageBatch)
         })
 
+        // TODO talk with Honza about the removing group ids in batch
```
Comment (Crhaj) [inline tests/consumer.spec.ts:992]: 👀

**Review by Crhaj — APPROVED** [review]

File: tests/consumer.spec.ts (hunk 2 above)
Comment (zabkwak) [reply tests/consumer.spec.ts:992]: sem to nesmazal? 🤦

File: src/consumer.ts (hunk 1 above)
Comment (zabkwak) [reply src/consumer.ts:394]: what? this is ES6 array destructuring

Comment (Crhaj) [reply src/consumer.ts:394]: furt hnusny

Comment (zabkwak) [reply src/consumer.ts:394]: 🙄

Comment (Crhaj) [reply src/consumer.ts:394]: 🥸

**Pull request merged**